core/services/vrf/listener_v2.go

```diff
@@ -560,7 +571,14 @@ func (lsn *listenerV2) processRequestsPerSubBatch(
 			}
 		}
 
-		fromAddress, err := lsn.gethks.GetRoundRobinAddress(lsn.chainID, lsn.fromAddresses()...)
+		fromAddresses := lsn.fromAddresses()
+		err = setMaxGasPriceGWei(fromAddresses, lsn.job.VRFSpec.MaxGasPriceGWei, lsn.keyUpdater, lsn.chainID)
```

Comment:
I guess we need this here because we want to support the use case where job spec is updated while node is running ?

Reply:
This is to protect against the max gas price somehow diverging while the job is running. We've seen this happen before, but we don't know the reason why. This basically ensures that the key-specific gas price is always set to the price specified in the job spec.

Reply:
If we re-create the job, this object (listenerV2) will end up getting recreated. Handling a new listenerV2 is in delegate.go.